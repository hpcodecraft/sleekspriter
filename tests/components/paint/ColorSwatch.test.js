import ColorSwatch from "components/paint/ColorSwatch";

describe("ColorSwatch", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<ColorSwatch />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render with a background color", () => {
    const wrapper = shallow(<ColorSwatch color="#ff0000" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should call an action handler with its color on click", () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <ColorSwatch color="#ff0000" action={clickHandler} />
    );
    wrapper.find(".colorswatch").simulate("click");
    expect(clickHandler).toBeCalledWith("#ff0000");
  });
});
