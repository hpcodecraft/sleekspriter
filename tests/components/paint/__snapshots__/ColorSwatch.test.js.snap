// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSwatch should render correctly 1`] = `"<div class=\\"colorswatch\\"></div>"`;

exports[`ColorSwatch should render with a background color 1`] = `"<div class=\\"colorswatch\\" style=\\"background:#ff0000;\\" title=\\"#ff0000\\"></div>"`;
